### ISSUE TITLE
Missing index for `NewExprField::Args`

### ISSUE THREAD

**Issue by hyf0** [swc-project/swc#6871]
### Describe the bug

# tl;dr

The definition of `NewExprField::Args` should be `NewExprField::Args(usize)` just like [`CallExprField::Args(usize)`](https://rustdoc.swc.rs/swc_ecma_visit/fields/enum.CallExprField.html).

# Backgroud

![img_v2_7450b0f5-cac7-4d89-a54b-488fe018c4eg](https://user-images.githubusercontent.com/49502170/215308770-90a4c52c-27b3-4a9c-bbfd-797b34f2ba78.jpg)

For `require('./types/shared')` and `require('./types/other')`, they have the same parent path. 

This causes the problem while trying to rewrite the `require('xxx')` expression. We couldn't distinguish the two expressions.

![d2581d3a-65a9-4c78-ac2b-f059f1577c4c](https://user-images.githubusercontent.com/49502170/215308880-9b5b5587-5f70-4a17-9705-c621e3260ab1.jpeg)

### Expected behavior

_

### Version

latest

### Additional context

I reproduce the issue on Turbopack, and it has the problem too.
